server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/AllocationsController.java

```diff
@@ -0,0 +1,134 @@
+ * Created by ywkim on 3/14/2022.
+ */
+
+@SwaggerDefinition(
+    info = @Info(
+        description = "IN-CORE Allocations Service for getting the user's allocation information.",
+        version = "v0.1",
+        title = "IN-CORE v2 Allocation Service API",
+        contact = @Contact(
+            name = "IN-CORE Dev Team",
+            email = "incore-dev@lists.illinois.edu",
+            url = "https://incore.ncsa.illinois.edu"
+        ),
+        license = @License(
+            name = "Mozilla Public License 2.0 (MPL 2.0)",
+            url = "https://www.mozilla.org/en-US/MPL/2.0/"
+        )
+    ),
+    consumes = {"application/json"},
+    produces = {"application/json"},
+    schemes = {SwaggerDefinition.Scheme.HTTP}
+
+)
+
+@Api(value = "allocations", authorizations = {})
+
+@Path("allocations")
+@ApiResponses(value = {
+    @ApiResponse(code = 500, message = "Internal Server Error")
+})
+
+public class AllocationsController {
+    @Inject
+    private IGroupAllocationsRepository allocationsRepository;
+
+    @Inject
+    private IUserFinalQuotaRepository finalQuotaRepository;
+
+    private static final Logger logger = Logger.getLogger(AllocationsController.class);
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @ApiOperation(value = "Gives the status of the service.",
+        notes = "This will provide the status of the service as a JSON.")
+    public String getStatus() {
+        String statusJson = "{\"status\": \"responding\"}";
+        return statusJson;
+    }
+
+    @GET
+    @Path("users/{username}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gives the allocation status of the given username.",
+        notes = "This will only work for admin user group.")
+    public String getAllocationsByUsername(
+        @HeaderParam("x-auth-userinfo") String userInfo,
+        @HeaderParam("x-auth-usergroup") String userGroup,
+        @ApiParam(value = "Dataset Id from data service", required = true) @PathParam("username") String userId) {
+        // check if the logged in user is in admin group
+        Boolean isAdmin = JsonUtils.isLoggedInUserAdmin(userGroup);
+
+        JSONObject outJson = new JSONObject();
+
+        if (isAdmin) {
+            try {
+                outJson = JsonUtils.createUserFinalQuotaJson(userId, finalQuotaRepository);
+            } catch (ParseException e) {
+                logger.error("Error extracting user status");
+                throw new IncoreHTTPException(Response.Status.INTERNAL_SERVER_ERROR, "Error extracting user status");
+            }
+        } else {
+            logger.error("Error extracting user status");
+            throw new IncoreHTTPException(Response.Status.FORBIDDEN, "Logged in user is not incore admin");
+        }
+
+        return outJson.toString();
+    }
+
+    @GET
+    @Path("groups/{groupname}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gives the allocation status of the given username.",
```

Comment:
should be: Gives the allocation status of the given group name

Reply:
changed